src/Directions.ts

```diff
@@ -5,5 +5,15 @@ export const Directions = {
   DOWN: 8,
 } as const;
 
-// eslint-disable-next-line @typescript-eslint/no-redeclare -- backward compatibility; it can be used as a type and as a value
+export const CornerDirections = {
```

Comment:
Now when I think of it, it could be slightly harder to implement `Vector.fromDirection`.

Reply:
The real issue i see is that minimal detection angle would be capped at >=30 percent, as previously Corners had a different, adaptive detection angle to fill the missing uncovered space, and after combining both classes, it no longer works that way.
I also think exposing directions like UP_RIGHT etc. to the user looks unclean.

Reply:
Nevermind, since current minimal angle is 40 degrees, and it is an unchangeable constant, i will apply all your suggestions.

Reply:
> That's a good call. I don't think it would be harder though, as fromDirection looks for exact matches, and i don't see edge-cases where it could return the wrong thing.

It does, but in case of `ALL` we don't have exact match 😄 

> But it makes Directions implementation less readable, it is not possible to reference this. nor Directions. when inside, thus we have to rely on normal numbers or a separate object.

We could make a structure like `BaseDirections`, `DiagonalDirections` and then combine them into `Directions`, though it could be too much.

> I also think exposing directions like UP_RIGHT etc. to the user looks unclean.

I agree, that's why I want to hear what @j-piasecki has to say in that case, as exposing things such as `UP_RIGHT` may or may not make sense (I think it would if we didn't have `|` operator to combine directions).

Reply:
> It does, but in case of ALL we don't have exact match 😄

We have an exact match for every single possible mask, because `fromDirection` is called for every `Directions` member, and evaluates it's individual activation. For example, if both `UP` and `RIGHT` are active, then all 3 `UP`, `RIGHT` and `UP_RIGHT` will trigger the `.some(Boolean)` individually. If we were to add an `ALL`, then while it wouldn't match by itself with anything, it would activate each of the other directions individually, guaranteeing `.some(Boolean)` to be activated.

Reply:
Ok but since we agree on:

> I also think exposing directions like UP_RIGHT etc. to the user looks unclean.

I think I can separate the exposed and the used api into `Directions`, `CompositeDirections` and `CombinedDirections`, of which only `Directions` will be exposed to the user.
Is that okay by you?